### DISCUSSION TITLE
Version 119.4.30 works on Azure App Service

### DISCUSSION THREAD

**Discussion by Ravnholt2102** [cefsharp/CefSharp#4776] [General]
A simple Web Api using CefSharp.OffScreen version 119.4.30 works when uploaded to azure app service, but using an older or newer version fails.
Looking at the changelog for CefSharp, it seems version 119.4.30 only updated chrome with a security update, so i don't see how that should enable support for Azure App Service over the previous version.

**Comment by amaitland** [comment]
Not something I've ever needed, so I can only provide generic debugging tips. I'm actually surprised any version works to be honest.

Make sure you have `GPU Acceleration and Compositing` disabled.

Check the log file for errors. https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file

**Comment by delta42** [comment]
Is CEF even supported on Azure? According to the [Azure Web App sandbox limitations](https://github.com/projectkudu/kudu/wiki/Azure-Web-App-sandbox), User32/GDI32 are heavily restricted. Granted those limitations are rather vague! And I'm not even sure what CEF uses User32 for.

My experiments are limited to trying out versions 119.4.30, 120.2.7 and 123.0.60 on Azure Web App, but in all cases Cef.Initialize returns false, without any logs giving any hint of what's going on.

With versions 20.2.7 and 123.0.60 I also get a crash with this additional detail:

`ERROR:main_runner.cc(251)] ContentMainRun failed with exit code 21`

Is this [CEF_RESULT_CODE_PROFILE_IN_USE](https://github.com/chromiumembedded/cef/blob/master/include/internal/cef_types.h#L1075)? Another dead end.

Scouring the internet for hours, I can neither find proof of anyone's success in using CEF with Azure App Service (Web App or Functions), nor 
a definitive statement about CEF's incompatibility with it.

**Comment by amaitland** [comment]
Follow up discussion at https://www.magpcss.org/ceforum/viewtopic.php?f=6&t=19763

Given this isn't a `CefSharp` specific problem, let's continue the conversation there.
